src/handlers/delete_client.rs

```diff
@@ -1,23 +1,39 @@
 use {
     crate::{
         decrement_counter,
-        error::Result,
-        handlers::{Response, DECENTRALIZED_IDENTIFIER_PREFIX},
+        error::{Error::InvalidAuthentication, Result},
+        handlers::{authenticate_client, Response, DECENTRALIZED_IDENTIFIER_PREFIX},
         log::prelude::*,
         state::AppState,
     },
-    axum::extract::{Path, State as StateExtractor},
+    axum::{
+        extract::{Path, State as StateExtractor},
+        http::HeaderMap,
+    },
+    relay_rpc::domain::ClientId,
     std::sync::Arc,
 };
 
 pub async fn handler(
     Path((tenant_id, id)): Path<(String, String)>,
     StateExtractor(state): StateExtractor<Arc<AppState>>,
+    headers: HeaderMap,
 ) -> Result<Response> {
     let id = id
         .trim_start_matches(DECENTRALIZED_IDENTIFIER_PREFIX)
         .to_string();
 
+    let client_to_be_deleted = ClientId::new(id.clone().into());
+    if !authenticate_client(headers, &state.config.public_url, |client_id| {
```

Comment:
`public_url` needs the `https://` or `http://` prefix now. Same for all other function calls

Reply:
Wouldn't this line do that? https://github.com/WalletConnect/echo-server/blob/8f592e4253bee4cb34cf4de7563e80da2338f32f/terraform/ecs/main.tf#L73